src/crimson/admin/osd_admin.cc

```diff
@@ -248,40 +248,44 @@ class DumpMetricsHook : public AdminSocketHook {
     oss_name << ")";
     auto name = oss_name.str();
 
-    switch (auto v = metric(); v.type()) {
-    case data_type::GAUGE:
-      f->dump_float(name, v.d());
-      break;
-    case data_type::COUNTER:
-      [[fallthrough]];
-    case data_type::DERIVE:
-      f->dump_unsigned(name, v.ui());
-      break;
-    case data_type::HISTOGRAM: {
-      f->open_object_section(name);
-      auto&& h = v.get_histogram();
-      f->dump_float("sum", h.sample_sum);
-      f->dump_unsigned("count", h.sample_count);
-      f->open_array_section("buckets");
-      for (auto i : h.buckets) {
-        f->open_object_section("bucket");
-        f->dump_float("le", i.upper_bound);
-        f->dump_unsigned("count", i.count);
-        f->close_section(); // "bucket"
-      }
-      {
-        f->open_object_section("bucket");
-        f->dump_string("le", "+Inf");
+    try {
+      switch (auto v = metric(); v.type()) {
+      case data_type::GAUGE:
+        f->dump_float(name, v.d());
+        break;
+      case data_type::COUNTER:
+        [[fallthrough]];
+      case data_type::DERIVE:
+        f->dump_unsigned(name, v.ui());
```

Comment:
@cyx1231st i think we should fix it instead, see #42203

Reply:
Oh, thanks, should we keep the try-catch as a fall-back to dump metircs?

Reply:
no, i'd just crash the application. as i think the mapping from datatype and the underlying type in the `variant<>` is a contract between the application (crimson) and the library (seastar). if the contract is broken, that's a bug.

Reply:
commit dropped.